tests/bootstrap_tests/test_resolver.py

```diff
@@ -178,43 +194,46 @@ def test_plugin_id_matching(self):
             bundle_cache_fallback_paths=[self.install_root]
         )
 
+        def _match_plugin_helper(plugin_ids):
```

Comment:
Instead of having a helper function, I wonder if we should instead have a loop here. Something like

```python
test_cases = [("foo.maya", [("", False), ("None", False)...]), ...]
for expected_plugin, (plugin_id, expected_result) in test_cases:
    resolver._plugin_id = expected_plugin
    self.assertEqual(resolver._matches_current_plugin_id({"plugin_ids": plugin_ids}), expected_result)
```

I'm  fine with either to be honest. It might be more readable the way it's written at the moment.